### PR TITLE
REF: generate setter action

### PR DESCRIPTION
**Pull request by Kobzol**
This PR adds an action to generate a setter for a struct, mirroring the recently added generate getter action (https://github.com/intellij-rust/intellij-rust/pull/5352).

setter

It also introduces several improvements like better copy type handling, support for type aliases, finding existing impl blocks and navigation to the generated method. Since these are directly applicable to both getters and setters, I included them in this PR, but if you want, I can split them into a separate PR.

Currently, if the action is invoked inside an `impl`, I filter out attributes that would conflict with an existing method:
```rust
struct S { a: u32 }
impl S {
   fn a(&self) {}
   /*caret*/ <- a is not offered, because `a` method exists
}
```
However, if the action is invoked on the struct, all (non-pub) fields are offered, and then if it finds an existing `impl` block with a method with the same name, it will generate the getter/setter and the names will conflict. I think that this behaviour is fine, just wanted to mention it.

### CODE REVIEW

File: src/main/kotlin/org/rust/ide/refactoring/generate/BaseGenerateHandler.kt [hunk 1]
```diff
@@ -95,8 +97,16 @@ abstract class BaseGenerateHandler : LanguageCodeInsightActionHandler {
         struct: RsStructItem
     ): RsImplItem {
         return if (implBlock == null) {
-            val impl = psiFactory.createInherentImplItem(structName, struct.typeParameterList, struct.whereClause)
-            struct.parent.addAfter(impl, struct) as RsImplItem
+            val existingImpl = struct.implLookup.findImplsAndTraits(struct.declaredType)
+                .filterIsInstance<TraitImplSource.ExplicitImpl>()
+                .filter { it.isInherent }
+                .firstOrNull()
```
Comment (vlad20012) [inline src/main/kotlin/org/rust/ide/refactoring/generate/BaseGenerateHandler.kt:103]: This `firstOrNull` looks very suspicious. A struct can have multiple inherent impls. Or impls with different type bounds... There can be one another dialog that allows to choose the impl (or choose to create a new one). But a new dialog overloads the UI, so such a solution is also very discussable. I'd prefer to generate a new impl over this design.

**Review by vlad20012 — COMMENTED** [review]
LGTM except one commit

File: src/main/kotlin/org/rust/ide/refactoring/generate/BaseGenerateHandler.kt (hunk 1 above)
Comment (Kobzol) [reply src/main/kotlin/org/rust/ide/refactoring/generate/BaseGenerateHandler.kt:103]: Oh right, I didn't think of impls with different type bounds. Ok, let's ditch finding existing impls. Should I also remove it in https://github.com/intellij-rust/intellij-rust/pull/6039?

Comment (vlad20012) [reply src/main/kotlin/org/rust/ide/refactoring/generate/BaseGenerateHandler.kt]: Yep, I just reviewed #6039 and was just about to write a comment about it =D

**Comment by vlad20012** [comment]
:+1: 
bors r+

**Comment by vlad20012** [comment]
Oh, you forgot to remove `test find existing impl block` test.
bors r-

**Comment by Kobzol** [comment]
bors r=vlad20012

**Comment by vlad20012** [comment]
I'll cherry-pick it into `release-130` because it mostly consists of fixes and improvements for #5352 that is a part of `release-130`

**Comment by MarinaKalashina** [comment]
Docs done (mentioned as an example for the Generate menu) https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-code-generation.html#generate-menu

**Pull request merged**